it.polito.elite.dog.drivers.modbus.gateway/src/it/polito/elite/dog/drivers/modbus/gateway/ModbusGatewayDriver.java

```diff
@@ -218,8 +230,32 @@ public String attach(ServiceReference reference) throws Exception
 			Set<String> gatewayProtocolSet = ((ControllableDevice) this.context.getService(reference))
 					.getDeviceDescriptor().getSimpleConfigurationParams().get(ModbusInfo.PROTO_ID);
 			
+			// Get the values for the parameters for the serial port (valid only in case of serial connections
```

Comment:
This artificially increases the service use count on the gateway device. every getService call shall have a corresponding ungetService call, unless there is a specific purpose in leaving the service marked as "used". This is done during the gateway creation instance and should not be done elsewhere